test/test_testing.py

```diff
@@ -776,9 +784,16 @@ def test_update_config(self):
             harness.charm.changes,
             [{'name': 'config-changed', 'data': {'a': 'foo', 'b': 2}},
              {'name': 'config-changed', 'data': {'a': 'foo', 'b': 3}},
-             {'name': 'config-changed', 'data': {'a': ''}},
+             {'name': 'config-changed', 'data': {'a': '', 'b': None}},
```

Comment:
This is actually problematic, because it isn't how *juju* does it.
I just created:
```
$ cat config.yaml
options:
    s:
        type: string
        default: "abcd"
    t:
        type: string
```

And then did a deploy, and 
```
$ juju run --unit ubuntu-operator/3 config-get
s: abcd
```

So juju does, indeed, not return keys that are null. So I don't think we want to include them here.

Doing a couple of other tests:
```
$ juju run --unit ubuntu-operator/3 -- 'config-get t ; echo $?'
0
```
So juju just returns the empty string from 'config-get' but doesn't return an error. However it also lets you ask about any config item:
```
$ juju run --unit ubuntu-operator/3 -- 'config-get r ; echo $?'
0
```
It *doesn't* let you set them, though:
```
$ juju config ubuntu-operator r=1234
{}
ERROR parsing settings for application: unknown option "r"
```
(though we shouldn't be printing {} there...)

Setting them does cause them to show up:
```
$ juju config ubuntu-operator t=1234
$ juju run --unit ubuntu-operator/3 -- 'config-get ; echo $?'
s: abcd
t: "1234"
0
$ juju run --unit ubuntu-operator/3 -- 'config-get t ; echo $?'
1234
0
```

And unsetting them removes it:
```
$ juju config ubuntu-operator --reset t
$ juju run --unit ubuntu-operator/3 -- 'config-get ; echo $?'
s: abcd
0
$ juju run --unit ubuntu-operator/3 -- 'config-get t ; echo $?'
0
```